### DISCUSSION TITLE
CopyPaste not work

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#1008] [Q&A]
Hi,
    I want to add copy cut paste to my diagram, I follow some steps from the workflow example, and I use the node server. My model is a custom JSON file.
1.set enableCopyPaste to true
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/82bfb555-6774-4678-a5fd-0ed7bd848051)
2.add hanlder to my diagram module, the RequestClipboardDataActionHandler CutOperationHandler PasteOperationHandler come from '@eclipse-glsp/server-node'
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/1bd80c6a-402e-421b-8a85-a651451fb982)
3.register operation in my server
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/e1fb12e8-bf96-4f46-aa90-099c5f4d4143)
I don't know if the third step is needed. I used ctr+c and ctr+v to copy and paste, But It seems that the handler is not working. It is like the handle not bind successful
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/7f5f7316-a4ce-4cdc-ae57-dc9712dc67a4)
I try to debug the paste handler file but it always can not be debugged. 
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/3b816431-8a6c-4dd3-9125-b329e164ec58)
Did I miss something? any help, thank you very much

**Comment by wss29** [comment]
I find that because I use a custom model, the RequestClipboardDataActionHandler and PasteOperationHandler can not handle it well. I rewrite myClipboardDataActionHandler and myPasteOperationHandler it works.
but another problem is the context menu not work, only keyword ctrl+c and ctrl+v work
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/10db9966-8525-44bd-9c68-0dda95a3bc58)
I debug and find the problem in copy-paste-handler. the shouldCopy method check selectedElementIds, When I right click open context menu the selected elements count is 0,
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/4de3e571-cd66-4336-acb1-059874b7e145)

**Comment by tortmayr** [comment]
Hi @wss29 ,

Yes you are correct. Copy & pase can not be handled generically and has to be customized for your source model. The provided default handlers only work if your directly use the `GModel` as source model. So using custom handlers for copy & paste is definitely the right way to go.

Regarding the mentioned bug. I'm, going to try to reproduce the issue and further investigate the problem and will come back to you once I know more.
